kbs/src/api/src/http/resource.rs

```diff
@@ -123,13 +99,24 @@ pub(crate) async fn get_resource(
         .await
         .map_err(|e| Error::ReadSecretFailed(e.to_string()))?;
 
-    let jwe = jwe(pubkey, resource_byte)?;
-
-    let res = serde_json::to_string(&jwe).map_err(|e| Error::JWEFailed(e.to_string()))?;
-
-    Ok(HttpResponse::Ok()
-        .content_type("application/json")
-        .body(res))
+    // If public key is included in claims, use JWE to wrap resource byte,
+    // else check if HTTPS is enabled.
+    match get_pkey(claims) {
```

Comment:
If I understand correctly we want to support requests without public key for KBS deployments without HTTPs. In this case there would never be a be pkey, right? Does it still make sense to attempt to fetch it or should we rather start the with the HttpsEnabled == false check and return the unwrapped resource? (Instead of swallowing the get_pkey() error).

Reply:
As you mentioned, this PR hopes to support requests without pubkey in KBS with HTTPS enabled. However, "HTTPS enabled" is a minimum security measure. Even if HTTPS is enabled, pubkeys may still exist. In this case, the returned resource data is protected by both HTTPS and JWE. Only when the pubkey cannot be found, data will be returned when HTTPS is enabled. If HTTPS is not enabled, it is completely unsafe, and KBS will refuse to return data.

Reply:
I see. thanks for clarifying.  So, it looks like having no pubkey in the claims is a legitimate code path, i.e. the signature is actually something like: `fn get_pkey() => Option<Result<TeePubKey>>`. A result should capture unexpected states. An alternative would be to match on a concrete error type:

```rust
Err(err) => {
   if let err = Error:NoPubkeyInClaims && https.enabled {
   ....
   }
   Err(Error::InsecureResponse)
}
```

Failing to parse (` TeePubKey::deserialize(pkey_value)`) should probably result in an error and not in a fallback to https, because this shouldn't happen in any case.

If `customized_claims` is specified as always having `runtime_data.tee-pubkey` we probably want to error out, too.